dart/lib/src/sentry_client.dart

```diff
@@ -20,21 +33,22 @@ class SentryClient {
     if (options.transport is NoOpTransport) {
       options.transport = HttpTransport(options);
     }
-    return SentryClient._(options);
-  }
 
-  final SentryOptions _options;
-
-  final Random _random;
-
-  final SentryExceptionFactory _exceptionFactory;
-
-  static final _sentryId = Future.value(SentryId.empty());
+    return SentryClient._(
+      options,
+      stackTraceFactory: SentryStackTraceFactory(options),
```

Comment:
why dont you do this inside of the private ctor like the `SentryExceptionFactory`? seems that each field is inited in a different place.

Reply:
we need an existing instance to instantiate the SentryExceptionFactory

Reply:
we can't do 
```dart
SentryClient._(this._options)  : _stackTraceFactory = stackTraceFactory,
        _exceptionFactory = SentryExceptionFactory(
          options: _options,
          stacktraceFactory: _stackTraceFactory,
        ),
        _random = _options.sampleRate == null ? null : Random();
```

Reply:
can we do both assignments then in the method body?

Reply:
✅